api/src/main/scala/hmda/api/http/HttpApi.scala

```diff
@@ -2,18 +2,34 @@ package hmda.api.http
 
 import java.net.InetAddress
 import java.time.Instant
+import akka.Done
+import akka.actor.ActorSystem
 import akka.event.LoggingAdapter
 import akka.http.scaladsl.marshalling.ToResponseMarshallable
+import akka.http.scaladsl.model.{ HttpResponse, StatusCodes, Multipart }
+import akka.http.scaladsl.model.Multipart.BodyPart
 import akka.http.scaladsl.server.Directives._
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{ Sink, Framing }
+import akka.util.ByteString
 import hmda.api.model.Status
 import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+import hmda.api.processing.ProcessingActor
+import hmda.api.processing.ProcessingActor.StopActor
 import hmda.api.protocol.HmdaApiProtocol
 import spray.json._
 
+import scala.concurrent.Future
+import scala.util.{ Failure, Success }
+
 trait HttpApi extends HmdaApiProtocol {
 
+  implicit val system: ActorSystem
+  implicit val materializer: ActorMaterializer
   val log: LoggingAdapter
 
+  val splitLines = Framing.delimiter(ByteString("\n"), 2048, allowTruncation = true)
```

Comment:
What's `allowTruncation = true` all about?  I'm having a hard time finding a good definition of that argument in their docs.  I think this snippet from [Streaming TCP](http://doc.akka.io/docs/akka-stream-and-http-experimental/2.0.3/scala/stream-io.html#Streaming_TCP) doc describes it, but it's a little unclear:

>The last boolean argument indicates that we require an explicit line ending even for the last message before the connection is closed.

Is that what's going on here?